app/models/movie.rb

```diff
@@ -1,3 +1,21 @@
 class Movie < ApplicationRecord
   belongs_to :studio
+  has_many :actor_movies
+  has_many :actors, through: :actor_movies
+
+  validates :title, presence: true
+  validates :creation_year, presence: true
+  validates :genre, presence: true
+
+  def actors_sorted
+    self.actors.order(:age)
+  end
+
+  def actor_age_avg
+    if self.actors.count(:id) == 0
+      0
+    else
+      self.actors.average(:age).round(2)
+    end
+  end
```

Comment:
I just want to make this quick note here that depending on who you ask, doing something like a `round` here could be considered presentation logic. I won't mark this as an MVC infraction, but something I wanted to bring up because some rubyist would consider this an infraction because it is technically presentation logic, I am in that camp.